### PR TITLE
[FLOW-1827] upgrade python version for circleci

### PR DESCRIPTION
**Pull request by crispin-clark**

### CODE REVIEW

File: Dockerfile.circleci [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM circleci/python:3.7.4-node
+FROM circleci/python:3.8.10-node
```
Comment (ryanb93) [inline Dockerfile.circleci:1]: `circleci/python` has been deprecated in favour of `cimg/python` 

> This image is designed to supercede the legacy CircleCI Python image, circleci/python.

Should that be used instead while we're changing things?

Comment (ryanb93) [reply Dockerfile.circleci:1]: Just a warning that it uses `pyenv` so the location of any hardcoded references to `site-packages` will have changed, got caught out here: https://github.com/babylonhealth/cd-workflows/pull/236/files#diff-a5f97da102a4ba5379ea1191b94f835106a75b8202903a1aef8391cd36a089b6R10

**Review by gleaver — APPROVED** [review]
Happy with this, though we might as well change the base tag as Ryan suggests

File: Dockerfile.circleci (hunk 1 above)
Comment (crispin-clark) [reply Dockerfile.circleci]: checked and there don't seem to be any in manifests either

**Pull request merged**